source/extensions/filters/network/common/redis/client_impl.cc

```diff
@@ -137,15 +138,34 @@ void ClientImpl::onEvent(Network::ConnectionEvent event) {
 void ClientImpl::onRespValue(RespValuePtr&& value) {
   ASSERT(!pending_requests_.empty());
   PendingRequest& request = pending_requests_.front();
-  if (!request.canceled_) {
-    request.callbacks_.onResponse(std::move(value));
-  } else {
+
+  if (request.canceled_) {
     host_->cluster().stats().upstream_rq_cancelled_.inc();
+  } else if (config_.enableRedirection() && (value->type() == Common::Redis::RespType::Error)) {
+    std::vector<absl::string_view> err = StringUtil::splitToken(value->asString(), " ", false);
+    bool redirected = false;
```

Comment:
nit: move this variable inside the following if statement.

Reply:
The following "if (!redirected).." statement was improperly indented enclosed in the "if (err.size() == 3)" statement block (as the result of an editing mistake when refactoring down to a singled onRedirection() callback). I've verified this problem in a new redis_proxy integration test (BadRedirectStrings), and the code fix will be pushed up in the next update for this PR.